src/raw-value.ts

```diff
@@ -0,0 +1,304 @@
+            }
+        }
+    }
+
+    class ReverseProxyHandler implements ProxyHandler<ReverseProxyTarget> {
+        // original target for the proxy
+        private readonly target: ReverseProxyTarget;
+        // metadata about the shape of the target
+        private readonly meta: TargetMeta;
+
+        constructor(target: ReverseProxyTarget, meta: TargetMeta) {
+            this.target = target;
+            this.meta = meta;
+        }
+        initialize(shadowTarget: ReverseShadowTarget) {
+            const { meta } = this;
+            // adjusting the proto chain of the shadowTarget (recursively)
+            const rawProto = env.getRawValue(meta.proto);
+            ReflectSetPrototypeOf(shadowTarget, rawProto);
+            // defining own descriptors
+            copyReverseOwnDescriptors(shadowTarget, meta.descriptors);
+            // reserve proxies are always frozen
+            freeze(shadowTarget);
+            // future optimization: hoping that proxies with frozen handlers can be faster
+            freeze(this);
+        }
+        apply(shadowTarget: ReverseShadowTarget, thisArg: RawValue, argArray: RawValue[]): RawValue {
+            const { target } = this;
+            const secThisArg = env.getSecureValue(thisArg);
+            const secArgArray = env.getSecureValue(argArray);
+            let sec;
+            try {
+                sec = apply(target as SecureFunction, secThisArg, secArgArray);
+            } catch (e) {
+                // This error occurred when the outer realm attempts to call a
+                // function from the sandbox. By throwing a new raw error, we eliminates the stack
+                // information from the sandbox as a consequence.
+                let rawError;
+                const { message, constructor } = e;
+                try {
+                    // the error constructor must be a secure error since it occur when calling
+                    // a function from the sandbox.
+                    const sr: SecureRecord | undefined = WeakMapGet(env.som, constructor);
+                    // the raw constructor must be registered (done during construction of env)
+                    // otherwise we need to fallback to a regular error.
+                    rawError = construct(sr!.raw as RawFunction, [message]);
+                } catch (ignored) {
+                    // in case the constructor inference fails
+                    rawError = ErrorCreate(message);
+                }
+                throw rawError;
+            }
+            return env.getRawValue(sec);
+        }
+        construct(shadowTarget: ReverseShadowTarget, argArray: RawValue[], newTarget: RawObject): RawObject {
+            const { target: SecCtor } = this;
+            if (newTarget === undefined) {
+                throw TypeError();
+            }
+            const secArgArray = env.getSecureValue(argArray);
+            // const secNewTarget = env.getSecureValue(newTarget);
```

Comment:
debug code ☝️

Reply:
I don't remember why this was commented out in the previous version... let me think about it... why are we don't sending `secNewTarget` into the construct.

Reply:
it turns out that it is good! not sure why I commented this.